### ISSUE TITLE
Invalid pager payload can break the notebook 

### ISSUE THREAD

**Issue by takluyver** [jupyter/notebook#58]
If a kernel sends pager output like msgspec 4, without a mimebundle (IRkernel/IRkernel#112), the notebook gets in a state where you can't execute any more cells. This is a bug in IRkernel, but it shouldn't completely break the notebook UI.

From Firefox's JS Console when seeing the error:

```
Object { fileName: "http://localhost:8888/static/base/js/utils.js", lineNumber: 697, columnNumber: 18, stack: "WrappedError@http://localhost:8888/static/base/js/utils.js:697:19
reject/<@http://localhost:8888/static/base/js/utils.js:776:33
", message: "Couldn't process kernel message", error_stack: Array[2] } utils.js:777:21

Object { fileName: "http://localhost:8888/static/base/js/utils.js", lineNumber: 697, columnNumber: 18, stack: "WrappedError@http://localhost:8888/static/base/js/utils.js:697:19
reject/<@http://localhost:8888/static/base/js/utils.js:776:33
", message: "Couldn't process kernel message", error_stack: Array[7] } utils.js:777:21

Object { fileName: "http://localhost:8888/static/base/js/utils.js", lineNumber: 697, columnNumber: 18, stack: "WrappedError@http://localhost:8888/static/base/js/utils.js:697:19
reject/<@http://localhost:8888/static/base/js/utils.js:776:33
", message: "Couldn't process kernel message", error_stack: Array[7] } utils.js:777:21

Object { fileName: "http://localhost:8888/static/base/js/utils.js", lineNumber: 697, columnNumber: 18, stack: "WrappedError@http://localhost:8888/static/base/js/utils.js:697:19
reject/<@http://localhost:8888/static/base/js/utils.js:776:33
", message: "Couldn't process kernel message", error_stack: Array[7] } utils.js:777:21

Object { fileName: "http://localhost:8888/static/base/js/utils.js", lineNumber: 697, columnNumber: 18, stack: "WrappedError@http://localhost:8888/static/base/js/utils.js:697:19
reject/<@http://localhost:8888/static/base/js/utils.js:776:33
", message: "Couldn't process kernel message", error_stack: Array[7] } utils.js:777:21

Object { fileName: "http://localhost:8888/static/base/js/utils.js", lineNumber: 697, columnNumber: 18, stack: "WrappedError@http://localhost:8888/static/base/js/utils.js:697:19
reject/<@http://localhost:8888/static/base/js/utils.js:776:33
", message: "Couldn't process kernel message", error_stack: Array[7] }
```

**Comment by Carreau** [comment]
See jupyter/notebook#431 PR that try to fix it.

**Comment by jdfreder** [comment]
It's hard to test this if the bug doesn't exist in irkernel anymore.  @Carreau , we you able to test your merged PR against this?

**Comment by Carreau** [comment]
```
Kernel.prototype._handle_ws_message = function (e) {
        var that = this;
        this._msg_queue = this._msg_queue.then(function() {
            return serialize.deserialize(e.data);
        }).then(function(msg) {
+            if(msg.msg_type == "execute_reply" && (msg.content.payload[0]||{}).source == 'page'){
+                throw new Error("gnnnnaaa")
+            }
            return that._finish_ws_message(msg);
        })
        .catch(function(error) {console.error("Couldn't process kernel message", error) });
    };
```

Manually. Still searching how to test that.

**Comment by Carreau** [comment]
Do you think monkey patching `_finish_ws_message` ? executing a `something?` and un-patching ? Seem brittle.

**Comment by Carreau** [comment]
And not sure worth it.

**Comment by jdfreder** [comment]
No, it's probably not worth it.  I just wasn't sure what steps were required to reproduce the error manually.  Thanks for posting the above though.  I'll test that here and close the issue if it works in master.

**Comment by Carreau** [comment]
closed by #431 @jdfreder ?

**Comment by jdfreder** [comment]
Closed!
